src/library/basesqltablemodel.cpp

```diff
@@ -423,7 +423,7 @@ void BaseSqlTableModel::setSort(int column, Qt::SortOrder order) {
     int trackSourceColumnCount = m_trackSource ? m_trackSource->columnCount() : 0;
 
     if (column < 0 ||
-            column >= trackSourceColumnCount + m_sortColumns.size() - 1) {
+            column >= trackSourceColumnCount + m_tableColumns.size() + m_sortColumns.size() - 1) {
```

Comment:
I think you have found a popy/paste issue. m_sortColumns.size() is the sort history kept for the secondary sort. 
```suggestion
            column >= trackSourceColumnCount + m_tableColumns.size() - 1) {
```